### PR TITLE
[Fluent] Adding unit test for related resources handling

### PR DESCRIPTION
**Pull request by anuchandy**
1. Fix for not handing creatable NIC
2. Unit test for related resources

### CODE REVIEW

**Comment by azurecla** [comment]
Hi __@anuchandy__, I'm your friendly neighborhood Azure Pull Request Bot (You can call me AZPRBOT). Thanks for your contribution!
    <p>
        It looks like you're working at Microsoft (anuchan). If you're full-time, we DON'T require a contribution license agreement.
    </p>
    <p>
        If you are a vendor, DO please sign the electronic contribution license agreement. It will take 2 minutes and there's no faxing! https://cla.azure.com.
    </p>

TTYL, AZPRBOT;

File: src/ResourceManagement/Azure.Fluent/Fluent.Tests/Compute/VirtualMachineTests.cs [hunk 1]
```diff
@@ -70,5 +76,108 @@ public void CanCreateVirtualMachine()
                 resourceManager.ResourceGroups.DeleteByName(RG_NAME);
             }
         }
+
+        [Fact(Skip = "TODO: Convert to recorded tests")]
+        public void CanCreateVirtualMachinesAndRelatedResourcesInParallel()
+        {
+            string resourceGroupName = ResourceNamer.RandomResourceName("rgvmtest-", 20);
```
Comment (anudeepsharma) [inline src/ResourceManagement/Azure.Fluent/Fluent.Tests/Compute/VirtualMachineTests.cs]: use var

File: src/ResourceManagement/Azure.Fluent/Fluent.Tests/Compute/VirtualMachineTests.cs [hunk 2]
```diff
@@ -70,5 +76,108 @@ public void CanCreateVirtualMachine()
...
+                {
+                    INetwork createdNetwork = (INetwork)createdVirtualMachines.CreatedRelatedResource(networkCreatableKey);
+                    Assert.NotNull(createdNetwork);
+                    Assert.True(networkNames.Contains(createdNetwork.Name));
+                }
+
+                HashSet<string> publicIpAddressNames = new HashSet<string>();
+                for (int i = 0; i < count; i++)
+                {
+                    publicIpAddressNames.Add(string.Format("{0}-{1}", publicIpNamePrefix, i));
```
Comment (anudeepsharma) [inline src/ResourceManagement/Azure.Fluent/Fluent.Tests/Compute/VirtualMachineTests.cs]: $"{publicIpNamePrefix}-{i}"

**Review by anudeepsharma — APPROVED** [review]
Looks good

**Comment by anuchandy** [comment]
Thanks @anudeepsharma

**Pull request merged**